### PR TITLE
[SHELL32] Disable Auto Arrange menu item when in details or list view

### PR DESCRIPTION
**Pull request by bkarish**
## Purpose

Enables/disables the 'Auto Arrange' menu item based on folder view mode.

### CODE REVIEW

File: dll/win32/browseui/shellbrowser.cpp [hunk 1]
```diff
@@ -1726,6 +1728,18 @@ void CShellBrowser::UpdateViewMenu(HMENU theMenu)
         SetMenuItemInfo(theMenu, IDM_VIEW_TOOLBARS, FALSE, &menuItemInfo);
     }
     SHCheckMenuItem(theMenu, IDM_VIEW_STATUSBAR, fStatusBarVisible ? TRUE : FALSE);
+
+    /* Disable Auto Arrange when in details or list view */
+    if (SUCCEEDED(fCurrentShellView->QueryInterface(IID_PPV_ARG(IFolderView, &pfv))))
+    {
+        if  (SUCCEEDED(pfv->GetCurrentViewMode(&viewMode)))
```
Comment (HBelusca) [inline dll/win32/browseui/shellbrowser.cpp]: Too much space between "if" and opening parenthesis.

File: dll/win32/shell32/CDefViewBckgrndMenu.cpp [hunk 2]
```diff
@@ -199,6 +199,18 @@ CDefViewBckgrndMenu::QueryContextMenu(HMENU hMenu, UINT indexMenu, UINT idCmdFir
             EnableMenuItem(hMenuPart, FCIDM_SHVIEW_INSERTLINK, MF_BYCOMMAND | MF_GRAYED);
         }
 
+        /* Disable Auto Arrange when in details or list view */
+        CComPtr<IFolderView> pfv;
+        UINT viewMode = 0;
+        if (SUCCEEDED(IUnknown_QueryService(m_site, SID_IFolderView, IID_PPV_ARG(IFolderView, &pfv))))
+        {
+            if  (SUCCEEDED(pfv->GetCurrentViewMode(&viewMode)))
```
Comment (HBelusca) [inline dll/win32/shell32/CDefViewBckgrndMenu.cpp]: Ditto

File: dll/win32/shell32/CDefViewBckgrndMenu.cpp [hunk 3]
```diff
@@ -199,6 +199,18 @@ CDefViewBckgrndMenu::QueryContextMenu(HMENU hMenu, UINT indexMenu, UINT idCmdFir
...
+            {
+                if (viewMode == FVM_DETAILS || viewMode == FVM_LIST)
```
Comment (HBelusca) [inline dll/win32/shell32/CDefViewBckgrndMenu.cpp]: There is no "else" case here, contrary to above?

File: dll/win32/shell32/CDefViewBckgrndMenu.cpp (hunk 2 above)
Comment (bkarish) [reply dll/win32/shell32/CDefViewBckgrndMenu.cpp]: Fixed in latest commit

File: dll/win32/browseui/shellbrowser.cpp (hunk 1 above)
Comment (bkarish) [reply dll/win32/browseui/shellbrowser.cpp]: Fixed in latest commit

File: dll/win32/shell32/CDefViewBckgrndMenu.cpp (hunk 3 above)
Comment (bkarish) [reply dll/win32/shell32/CDefViewBckgrndMenu.cpp]: The menu resource is loaded every time the context menu is launched in CDefViewBckgrndMenu.cpp, so it only needs to be disabled if needed. Shellbrowser reuses the same  menu, so it needs to be enabled or disabled every time the menu is displayed.

**Comment by yagoulas** [comment]
If these menu items are added by yhe CDefView, it is the CDefView that disables them

**Comment by Zero3K** [comment]
I think this should be closed since it was recently committed.

**Pull request closed without merging**